tools/scxtop/src/perfetto_trace.rs

```diff
@@ -275,6 +275,31 @@ impl<'a> PerfettoTraceManager<'a> {
             });
     }
 
+    /// Adds events for the softirq entry/exit events.
+    pub fn on_softirq(&mut self, action: &SoftIRQAction) {
+        let _ = self
+            .ftrace_events
+            .entry(action.cpu)
+            .or_insert_with(Vec::new)
+            .extend({
+                let mut entry_ftrace_event = FtraceEvent::new();
+                let mut exit_ftrace_event = FtraceEvent::new();
+                let mut entry_event = SoftirqEntryFtraceEvent::new();
+                let mut exit_event = SoftirqExitFtraceEvent::new();
+                entry_event.set_vec(action.softirq_nr as u32);
+                exit_event.set_vec(action.softirq_nr as u32);
+
+                entry_ftrace_event.set_timestamp(action.entry_ts);
+                entry_ftrace_event.set_softirq_entry(entry_event);
+                entry_ftrace_event.set_pid(action.pid);
+                exit_ftrace_event.set_timestamp(action.exit_ts);
+                exit_ftrace_event.set_softirq_exit(exit_event);
+                exit_ftrace_event.set_pid(action.pid);
+
+                vec![entry_ftrace_event, exit_ftrace_event]
```

Comment:
nit: I think (untested) that you can do this with a tuple and avoid allocating.

```suggestion
                (entry_ftrace_event, exit_ftrace_event)
```

Reply:
Annoying, it seems the tuple doesn't implement `IntoIterator`:
```
error[E0277]: `(FtraceEvent, FtraceEvent)` is not an iterator
   --> tools/scxtop/src/perfetto_trace.rs:284:21
    |
284 |               .extend({
    |  ______________------_^
    | |              |
    | |              required by a bound introduced by this call
285 | |                 let mut entry_ftrace_event = FtraceEvent::new();
286 | |                 let mut exit_ftrace_event = FtraceEvent::new();
287 | |                 let mut entry_event = SoftirqEntryFtraceEvent::new();
...   |
299 | |                 (entry_ftrace_event, exit_ftrace_event)
    | |                 --------------------------------------- this tail expression is of type `(FtraceEvent, FtraceEvent)`
300 | |             });
    | |_____________^ `(FtraceEvent, FtraceEvent)` is not an iterator
    |
    = help: the trait `Iterator` is not implemented for `(FtraceEvent, FtraceEvent)`, which is required by `(FtraceEvent, FtraceEvent): IntoIterator`
    = note: required for `(FtraceEvent, FtraceEvent)` to implement `IntoIterator`
note: required by a bound in `extend`
```

Reply:
I am dumb... `.into_iter()` exists

Reply:
I was dumb! Of course it doesn't, the types can be different. An array should actually work.